client/main.lua

```diff
@@ -2,6 +2,69 @@ local config = require 'config.client'
 local sharedConfig = require 'config.shared'
 local VEHICLES = exports.qbx_core:GetVehiclesByName()
 
+---@enum ProgressColor
+local ProgressColor = {
+    GREEN = 'green.5',
+    YELLOW = 'yellow.5',
+    RED = 'red.5'
+}
+
+---@param percent number
+---@return string
+local function getProgressColor(percent)
+    if percent >= 75 then
+        return ProgressColor.GREEN
+    elseif percent > 25 then
+        return ProgressColor.YELLOW
+    end
+
+    return ProgressColor.RED
+end
+
+---@enum StateLabels
+local StateLabels = {
+    OUT = 0,
+    GARAGED = 1,
+    IMPOUND = 2,
+}
+
+---@param state number
+---@return string
+local function getStateLabel(state)
+    if StateLabels.OUT == state then
+        return Lang:t('status.out')
+    elseif StateLabels.GARAGED == state then
+        return Lang:t('status.garaged')
+    elseif StateLabels.IMPOUND == state then
+        return Lang:t('status.impound')
+    end
+
+    return 'Unknown'
+end
+
+---@enum VehicleCategory
+local VehicleCategory = {
+    all = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22},
+    car = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 17, 18, 19, 20, 22},
+    air = {15, 16},
+    sea = {14},
+}
```

Comment:
I see you copied and pasted this so feel free to not address, but this isn't an enum and shouldn't be marked as such. Additionally, this should probably be a "set" table<integer, boolean> for O(1) lookup times. checkVehicleClass() can then be replaced by a lookup of the set to simplify the code as well.

Reply:
You've lost me on the set for O(1) lookup times